actions/templates.go

```diff
@@ -137,51 +133,61 @@ func PostTemplate(c *gin.Context) {
 		return
 	}
 
-	store, err := templates.NewSesTemplateStore(keys.AccessKey, keys.SecretKey, keys.Region)
-	if err != nil {
-		logger.From(c).WithError(err).Error("Unable to create SES template store.")
-		c.JSON(http.StatusBadRequest, gin.H{
-			"message": "SES keys are incorrect.",
+	template := &entities.Template{
+		UserID:      u.ID,
+		Name:        body.Name,
+		HTMLPart:    body.HTMLPart,
+		TextPart:    body.TextPart,
+		SubjectPart: body.Subject,
+	}
+
+	_, err := storage.GetTemplateByName(c, template.Name, u.ID)
+	if err == nil {
+		c.JSON(http.StatusUnprocessableEntity, gin.H{
+			"message": "Template with that name already exists",
 		})
 		return
 	}
 
-	_, err = store.CreateTemplate(&ses.CreateTemplateInput{
-		Template: &ses.Template{
-			TemplateName: aws.String(body.Name),
-			HtmlPart:     aws.String(body.Content),
-			TextPart:     aws.String(body.Content),
-			SubjectPart:  aws.String(body.Subject),
-		},
-	})
+	err = service.AddTemplate(c, template)
 	if err != nil {
-		reason := "Unable to create template."
-
-		if awsErr, ok := err.(awserr.Error); ok {
-			reason = fmt.Sprintf("%s: %s", awsErr.Code(), awsErr.Message())
+		switch {
+		case errors.Is(err, templatesvc.ErrParseHTMLPart):
+			c.JSON(http.StatusBadRequest, gin.H{
+				"message": "Unable to create template, failed to parse html_part",
+			})
+		case errors.Is(err, templatesvc.ErrParseTextPart):
+			c.JSON(http.StatusBadRequest, gin.H{
+				"message": "Unable to create template, failed to parse text_part",
+			})
+		case errors.Is(err, templatesvc.ErrParseSubjectPart):
+			c.JSON(http.StatusBadRequest, gin.H{
+				"message": "Unable to create template, failed to parse subject_part",
+			})
+		default:
+			logger.From(c).WithFields(logrus.Fields{
+				"template": template,
+				"user_id":  u.ID,
+			}).WithError(err).Error("Unable to create template")
+			c.JSON(http.StatusBadRequest, gin.H{
+				"message": "Unable to create template, please try again.",
+			})
 		}
-
-		c.JSON(http.StatusBadRequest, gin.H{
-			"message": reason,
-		})
 		return
 	}
 
-	c.JSON(http.StatusCreated, entities.Template{
-		Name:        body.Name,
-		HTMLPart:    body.Content,
-		TextPart:    body.Content,
-		SubjectPart: body.Subject,
-	})
+	c.JSON(http.StatusCreated, template)
 }
 
 func PutTemplate(c *gin.Context) {
 	u := middleware.GetUser(c)
+	service := templatesvc.NewTemplateService()
+	name := c.Param("name")
 
-	keys, err := storage.GetSesKeys(c, u.ID)
+	template, err := storage.GetTemplateByName(c, name, u.ID)
```

Comment:
I'm not sure if we need this check..the name can be changed to a non-existent one, we should only do the check below right?

Reply:
this is the name from path param the change-name is in body @FilipNikolovski

Reply:
We can change the path to be the template's ID, it will be the same as the other resources. It's ok if we have a breaking change here, we'll just update it on the UI side.

Reply:
Okay, in the meeting we discussed to remain name  as path param, but I will make the change. @FilipNikolovski

Reply:
@FilipNikolovski  resolved?